File: app/lib/core/models/backup.dart
enum BackupStatus { pending, uploading, uploaded, failed }

enum BackupTrigger { auto, manual }

class Backup {
  final String id;
  final String gameId;
  final String driveFileId;
  final String? driveRevId;
  final String fileName;
  final int sizeBytes;
  final int compressedBytes;
  final String checksum;
  final String? label;
  final BackupTrigger trigger;
  final BackupStatus status;
  final DateTime createdAt;

  const Backup({
    required this.id,
    required this.gameId,
    required this.driveFileId,
    this.driveRevId,
    required this.fileName,
    this.sizeBytes = 0,
    this.compressedBytes = 0,
    this.checksum = '',
    this.label,
    this.trigger = BackupTrigger.auto,
    this.status = BackupStatus.uploaded,
    required this.createdAt,
  });

  factory Backup.fromJson(Map<String, dynamic> json) => Backup(
        id: json['id'] as String,
        gameId: json['gameId'] as String,
        driveFileId: json['driveFileId'] as String,
        driveRevId: json['driveRevId'] as String?,
        fileName: json['fileName'] as String,
        sizeBytes: (json['sizeBytes'] as num?)?.toInt() ?? 0,
        compressedBytes: (json['compressedBytes'] as num?)?.toInt() ?? 0,
        checksum: json['checksum'] as String? ?? '',
        label: json['label'] as String?,
        trigger: (json['trigger'] as String?) == 'manual'
            ? BackupTrigger.manual
            : BackupTrigger.auto,
        status: _statusFromString(json['status'] as String?),
        createdAt: DateTime.tryParse(json['createdAt'] as String? ?? '') ??
            DateTime.now(),
      );

  static BackupStatus _statusFromString(String? s) {
    switch (s) {
      case 'uploading':
        return BackupStatus.uploading;
      case 'failed':
        return BackupStatus.failed;
      case 'pending':
        return BackupStatus.pending;
      default:
        return BackupStatus.uploaded;
    }
  }
}

File: backend/src/main/java/com/gameon/model/Backup.java
package com.gameon.model;

import jakarta.persistence.*;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "backups")
public class Backup {

    @Id
    @GeneratedValue
    private UUID id;

    @Column(name = "game_id", nullable = false)
    private UUID gameId;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(name = "drive_file_id", nullable = false)
    private String driveFileId;

    @Column(name = "drive_rev_id")
    private String driveRevId;

    @Column(name = "file_name", nullable = false)
    private String fileName;

    @Column(name = "file_path", nullable = false)
    private String filePath;

    @Column(name = "size_bytes")
    private Long sizeBytes = 0L;

    @Column(name = "compressed_bytes")
    private Long compressedBytes = 0L;

    private String checksum;

    private String label;

    @Column(name = "trigger_type")
    private String triggerType = "auto"; // auto | manual

    private String status = "pending"; // pending | uploading | uploaded | failed

    @Column(name = "created_at")
    private Instant createdAt;

    @PrePersist
    void onCreate() {
        createdAt = Instant.now();
    }

    // ── Getters / setters ────────────────────────────────────────────────
    public UUID getId() { return id; }
    public void setId(UUID id) { this.id = id; }

    public UUID getGameId() { return gameId; }
    public void setGameId(UUID gameId) { this.gameId = gameId; }

    public UUID getUserId() { return userId; }
    public void setUserId(UUID userId) { this.userId = userId; }

    public String getDriveFileId() { return driveFileId; }
    public void setDriveFileId(String driveFileId) { this.driveFileId = driveFileId; }

    public String getDriveRevId() { return driveRevId; }
    public void setDriveRevId(String driveRevId) { this.driveRevId = driveRevId; }

    public String getFileName() { return fileName; }
    public void setFileName(String fileName) { this.fileName = fileName; }

    public String getFilePath() { return filePath; }
    public void setFilePath(String filePath) { this.filePath = filePath; }

    public Long getSizeBytes() { return sizeBytes; }
    public void setSizeBytes(Long sizeBytes) { this.sizeBytes = sizeBytes; }

    public Long getCompressedBytes() { return compressedBytes; }
    public void setCompressedBytes(Long compressedBytes) { this.compressedBytes = compressedBytes; }

    public String getChecksum() { return checksum; }
    public void setChecksum(String checksum) { this.checksum = checksum; }

    public String getLabel() { return label; }
    public void setLabel(String label) { this.label = label; }

    public String getTriggerType() { return triggerType; }
    public void setTriggerType(String triggerType) { this.triggerType = triggerType; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    public Instant getCreatedAt() { return createdAt; }
    public void setCreatedAt(Instant createdAt) { this.createdAt = createdAt; }
}
